cpp/src/arrow/sparse_tensor_test.cc

```diff
@@ -910,4 +911,242 @@ TEST_F(TestSparseCSCMatrix, TestToTensor) {
   ASSERT_TRUE(tensor.Equals(*dense_tensor));
 }
 
+template <typename IndexValueType>
+class TestSparseCSFTensorBase : public ::testing::Test {
+ public:
+  void SetUp() {
+    dim_names_ = {"a", "b", "c", "d"};
+    shape_ = {2, 3, 4, 5};
+    int16_t dense_values[2][3][4][5] = {};  // zero-initialized
+
+    dense_values[0][0][0][1] = 1;
+    dense_values[0][0][0][2] = 2;
+    dense_values[0][1][0][0] = 3;
+    dense_values[0][1][0][2] = 4;
+    dense_values[0][1][1][0] = 5;
+    dense_values[1][1][1][0] = 6;
+    dense_values[1][1][1][1] = 7;
+    dense_values[1][1][1][2] = 8;
+
+    auto dense_buffer = Buffer::Wrap(dense_values, sizeof(dense_values));
+    Tensor dense_tensor_(int16(), dense_buffer, shape_, {}, dim_names_);
+    ASSERT_OK_AND_ASSIGN(
+        sparse_tensor_from_dense_,
+        SparseCSFTensor::Make(dense_tensor_,
+                              TypeTraits<IndexValueType>::type_singleton()));
+  }
+
+ protected:
+  std::vector<int64_t> shape_;
+  std::vector<std::string> dim_names_;
+  std::shared_ptr<SparseCSFTensor> sparse_tensor_from_dense_;
+};
+
+class TestSparseCSFTensor : public TestSparseCSFTensorBase<Int64Type> {};
+
+template <typename IndexValueType>
+class TestSparseCSFTensorForIndexValueType
+    : public TestSparseCSFTensorBase<IndexValueType> {
+ protected:
+  std::shared_ptr<SparseCSFIndex> MakeSparseCSFIndex(
+      const std::vector<int64_t> axis_order,
+      std::vector<std::vector<typename IndexValueType::c_type>>& indptr_values,
```

Comment:
`const&` for const vectors.

Reply:
You don't seem to mutate those vector inputs, so should make them a const reference really :-)